content/articles/todolist-with-svelte-and-firebase/index.md

```diff
@@ -0,0 +1,1135 @@
+  
+  router.start();
+</script>
+  
+<svelte:component this={page} />
+```
+
+We're importing the router, adding our different routes and starting the router. Our pages are empty but before adding their content, we need to set up our store. In the `src` folder, create a `stores.js`file and paste the following into it:
+
+```js
+import { writable } from "svelte/store";
+
+export const TodoStore = writable([]);
+
+export const UserStore = writable(null);
+
+```
+
+`writable` helps us create a store that we can edit down the line. We then create 2 new stores, a `TodoStore` and a `UserStore` to hold the list of todos and the user's details respectively. All `writable` stores have 2 main methods:
+
+- Set: To set the value of the store.
+- Update: To update the value of the store based on a callback function.
+
+Open the `login.svelte` file and paste:
+
+```js
+<script>
+  import { UserStore } from "./../stores";
+  import page from "page";
+
+  $: if ($UserStore) page.redirect("/");
+</script>
+
+<div class="login container">
+  <button>Log In</button>;
+</div>
+
+<style lang="scss">
+  div.login.container {
+    background: hsla(238, 100%, 71%, 1);
+
+    background: linear-gradient(
+      135deg,
+      hsla(238, 100%, 71%, 1) 0%,
+      hsla(295, 100%, 84%, 1) 100%
+    );
+
+    background: -moz-linear-gradient(
+      135deg,
+      hsla(238, 100%, 71%, 1) 0%,
+      hsla(295, 100%, 84%, 1) 100%
+    );
+
+    background: -webkit-linear-gradient(
+      135deg,
+      hsla(238, 100%, 71%, 1) 0%,
+      hsla(295, 100%, 84%, 1) 100%
+    );
+    min-height: 100vh;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    position: relative;
+    button {
+      background-color: hsla(350, 100%, 69%, 1);
+      height: 200px;
+      width: 200px;
+      box-shadow: -10px -10px 15px rgb(255, 119, 142),
+        10px 10px 15px rgb(231, 88, 112);
+      border: none;
+      border-radius: 50%;
+      border: 8px solid hsla(350, 100%, 69%, 1);
+      font-weight: bold;
+      color: white;
+      font-size: 2rem;
+      &:focus {
+        box-shadow: -10px -10px 15px rgb(255, 119, 142),
+          10px 10px 15px rgb(231, 88, 112),
+          inset -10px -10px 15px rgb(255, 119, 142),
+          inset 10px 10px 15px rgb(231, 88, 112);
+      }
+    }
+  }
+</style>
+```
+
+Here, we're displaying a large centred button, that will log the user in when clicked. The `$UserStore` is used to access the value inside of the store.
+
+In Svelte, statements written after `$:` like so `$: if ($UserStore) page.redirect("/")` are called `Reactive Statements`. They are run before a component updates if the values they depend on have changed. The \$ in `$UserStore` and `Reactive Statements` do not have the same function.
+
+The reactive statement in our component checks if the `$UserStore` has a value i.e if we have a user currently logged in and redirects to the homepage if it does.
+
+Paste this in the `home.svelte` file:
```

Comment:
Explain what do we have here?

Do this for every file that you use. Explain the purpose